galloc/docs/optimisation.md

```diff
@@ -0,0 +1,320 @@
+# Optimisation of galloc
+We've tried to optimise galloc in some ways, but none of them showed significant performance improvement, so we'd **strongly suggest to keep the current implementation.**
+
+The current implementation uses static buffer in addition to standard <kbd>dlmalloc</kbd>-like implementation.
+
+![Total perfomance comparison](./images/total-performance-comparison.png)
+
+Here are some of the optimisation attempts we've tried:
+
+1. [Measuring current performance](#current)
+1. [Remove static buffer](#remove)
+1. [Increase static buffer size by two](#increase)
+1. [Multiple static buffers](#multiple)
+
+## 0. Measuring current performance
+<a name="current"></a>
+We've measured the current [(commit 9135baa)](https://github.com/gear-tech/dlmalloc-rust/tree/9135baa728ef9a9a04a887998e019733c4b093af) performance of dlmalloc to have a baseline for comparison.
+### How we measured
+
+All optimisation attempts were tested 3 times on the different test cases, and the average of the results was used to compare the performance. Our main metric is gas consumption: the less gas is consumed, the better. All optimisations were measured in release mode binaries with most optimisations enabled and compared to current perfomance.
+
+The gas measurements were made via `debug` and `gas_available` syscalls, which was temporarily made free. Each allocation top-level operation (i.e. malloc, calloc, ...) was measured separately, the gas was measured before and after function call and calculated as substraction result of gas before and gas after.
+
+Then the gas consumption was calculated as a percentage of total gas spent on the test case.
+
+We've intentionally did not put any info about testing machine, because we measure only gas consumption, which will be same for every machine, since we have fixed weights for every instruction and syscall.
```

Comment:
```suggestion
We've intentionally not put any info about the testing machine because we measure only gas consumption, which will be the same for every machine since we have fixed weights for every instruction and syscall.
```